packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/collect_multi_namespace_references.ts

```diff
@@ -220,196 +227,3 @@ async function getObjectsAndReferences({
 
   return { objectMap, inboundReferencesMap };
 }
-
-/**
- * Checks/enforces authorization, writes audit events, filters the object graph, and redacts spaces from the share_to_space/bulk_get
- * response. In other SavedObjectsRepository functions we do this before decrypting attributes. However, because of the
- * share_to_space/bulk_get response logic involved in deciding between the exact match or alias match, it's cleaner to do authorization,
- * auditing, filtering, and redaction all afterwards.
- */
-async function optionallyUseSecurity(
```

Comment:
All authz logic is moved into the extension's authorizeAndRedactMultiNamespaceReferences method.